app/models/concerns/campaigns/presentable.rb

```diff
@@ -13,5 +13,21 @@ def duration
       return nil unless start_date && end_date
       distance_of_time_in_words(start_date, end_date, scope: "datetime.distance_in_words.short")
     end
+
+    def percentage_complete_by_date
+      return 0   unless start_date.present? && end_date.present?
+      return 0   if start_date > Date.current
+      return 100 if end_date < Date.current
+
+      ((remaining_days.to_f / total_days.to_f) * 100).to_i
+    end
+
+    def percentage_complete_by_budget
+      return 0   unless start_date.present? && end_date.present?
+      return 0   if start_date > Date.current
+      return 100 if end_date < Date.current
+
+      ((remaining_days.to_f / total_days.to_f) * 100).to_i
```

Comment:
See comment above about divide by zero. Note the first guard could/should be replaced with the `total_days` impl.